cmd/dra-example-kubeletplugin/main.go

```diff
@@ -159,18 +162,32 @@ func StartPlugin(ctx context.Context, config *Config) error {
 		return fmt.Errorf("path for cdi file generation is not a directory: '%v'", err)
 	}
 
-	driver, err := NewDriver(ctx, config)
-	if err != nil {
-		return err
-	}
+	var driver *driver
+	{
+		// A new context only for NewDriver. We don't want to use an
+		// already-done context after that finishes.
+		ctx, stop := signal.NotifyContext(ctx, syscall.SIGHUP, syscall.SIGINT, syscall.SIGTERM, syscall.SIGQUIT)
+		defer stop()
```

Comment:
```suggestion
		defer stop()
		ctx = klog.NewContext(ctx, logger)
```

Reply:
Does this only add the logger to the context? Is the parent's logger not inherited?

Reply:
Please ignore my comment. The `NotifyContext` parent context already has the logger.